modules/seqexec/server/src/main/scala/seqexec/server/SeqTranslate.scala

```diff
@@ -88,15 +90,15 @@ class SeqTranslate(site: Site, systems: Systems[IO], settings: TranslateSettings
     // endObserve must be sent to the instrument too.
     def notifyObserveEnd: SeqAction[Unit] = (inst +: otherSys).map(_.notifyObserveEnd).sequence.void
 
-    def closeImage(id: ImageFileId): SeqAction[Unit] =
-      inst.keywordsClient.closeImage(id)
+    def closeImage[F[_]: LiftIO](id: ImageFileId): SeqActionF[F, Unit] =
+      SeqActionF.embed(inst.keywordsClient.closeImage(id))
 
     def doObserve(fileId: ImageFileId): SeqAction[Result] =
       for {
         d   <- dataId
         _   <- sendDataStart(obsId, fileId, d)
         _   <- notifyObserveStart
-        _   <- headers(ctx).map(_.sendBefore(obsId, fileId)).sequence
+        _   <- headers(ctx).map(_.sendBefore(obsId, fileId)).sequence.embed
```

Comment:
What does `embed` do?

Reply:
embed will take a `F[_]` and convert it to `SeqActionF[_]`. It is like a boundary to avoid propagating the `F[_]` change upstream in the code (yet 😄 )

It also will preserve the error handling semantics